### PR TITLE
Use verbose style when suggesting changing `const` with `let`

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

**Review by compiler-errors — APPROVED** [review]

**Comment by ehuss** [comment]
@bors r-

Failed in https://github.com/rust-lang/rust/pull/127394#issuecomment-2211485337

**Comment by workingjubilee** [comment]
Hmm, that test needs to be minicored, I think.

File: tests/ui/const-generics/legacy-const-generics-bad.stderr [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
   --> $DIR/legacy-const-generics-bad.rs:7:35
    |
-LL |     let a = 1;
-   |     ----- help: consider using `const` instead of `let`: `const a`
 LL |     legacy_const_generics::foo(0, a, 2);
    |                                   ^ non-constant value
+   |
+help: consider using `const` instead of `let`
+   |
+LL |     const a = 1;
```
Comment (compiler-errors) [inline tests/ui/const-generics/legacy-const-generics-bad.stderr:9]: Wait, this suggestion is definitely wrong. Please mark it as *not* `MachineApplicable`.

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Also the suggestion needs to be marked `HasPlaceholders` or otherwise tweaked since it's not `MachineApplicable` in its current state.

File: tests/ui/const-generics/legacy-const-generics-bad.stderr (hunk 1 above)
Comment (compiler-errors) [reply tests/ui/const-generics/legacy-const-generics-bad.stderr:9]: Sorry, I should've noticed it sooner. I'll look at your diffs closer next time, but maybe you should look at them closer too 😉 Group effort 🙌

Comment (estebank) [reply tests/ui/const-generics/legacy-const-generics-bad.stderr:9]: The sigh was at the suggestion being wrong in the first place :)

**Review by compiler-errors — APPROVED** [review]

**Comment by compiler-errors** [comment]
r=me when ci green

**Comment by estebank** [comment]
@bors r-

I hadn't addressed https://github.com/rust-lang/rust/pull/127382#issuecomment-2211495007 yet.

**Comment by estebank** [comment]
@bors r=compiler-errors rollup=never

(In case there are more latent failures in other platforms.)

**Comment by tgross35** [comment]
Looks like maybe one extra line was missing (failed rollup https://github.com/rust-lang/rust/pull/127529)

```
024-07-09T19:22:37.6872320Z ---- [ui] tests/ui/asm/aarch64/parse-error.rs stdout ----
2024-07-09T19:22:37.6873300Z diff of stderr:
2024-07-09T19:22:37.6873600Z 
2024-07-09T19:22:37.6873920Z 398	LL |     const bar: /* Type */ = 0;
2024-07-09T19:22:37.6874630Z 399	   |     ~~~~~    ++++++++++++
2024-07-09T19:22:37.6875140Z 400	
2024-07-09T19:22:37.6875570Z -	
2024-07-09T19:22:37.6876020Z 402	error: aborting due to 57 previous errors
2024-07-09T19:22:37.6876560Z 403	
2024-07-09T19:22:37.6877170Z 404	For more information about this error, try `rustc --explain E0435`.
2024-07-09T19:22:37.6877690Z 
2024-07-09T19:22:37.6877810Z 
2024-07-09T19:22:37.6878100Z The actual stderr differed from the expected stderr.
```

**Comment by tgross35** [comment]
Ah this is testing again already, bors is moving too fast. Somebody should r-

**Pull request merged**